Frameworks/CoreGraphics/CGColor.mm

```diff
@@ -17,37 +17,154 @@
 #import <StubReturn.h>
 #import <Starboard.h>
 #import <CoreGraphics/CGColor.h>
-
-#import "UIColorInternal.h"
+#import <CFCppBase.h>
+#import "CGColorInternal.h"
 
 const CFStringRef kCGColorBlack = CFSTR("BLACK");
 const CFStringRef kCGColorWhite = CFSTR("WHITE");
 const CFStringRef kCGColorClear = CFSTR("CLEAR");
 
-static IWLazyClassLookup __LazyUIColor("UIColor");
-
 static const wchar_t* TAG = L"CGColor";
 
+struct __CGColor : CoreFoundation::CppBase<__CGColor> {
+    // Override to add Equals method
+    static CFTypeID GetTypeID() {
+        static __CFRuntimeClass _cls{
+            0, // Version
+            typeid(__CGColor).name(), // Class Name
+            nullptr, // Init
+            nullptr, // Copy (ill-defined for CF)
+            &_CFFinalize, // Finalize
+            &__CFEquals, // Equals
+            nullptr, // Hash
+            nullptr, // Copy Description
+            nullptr, // Copy Debug Description
+        };
+        static CFTypeID _id = _CFRuntimeRegisterClass(&_cls);
+        return _id;
+    }
+
+    __CGColor(CGColorSpaceRef colorspace, CGFloat r, CGFloat g, CGFloat b, CGFloat alpha) : _colorSpace(colorspace), _pattern(nullptr) {
+        _components.r = r;
+        _components.g = g;
+        _components.b = b;
+        _components.a = alpha;
+    }
+
+    __CGColor(CGColorSpaceRef colorSpace, CGPatternRef pattern) : _colorSpace(colorSpace), _pattern(pattern) {
+    }
+
+    // CreateCopyWithAlpha
+    __CGColor(CGColorRef color, float alpha) : _colorSpace(color->_colorSpace), _pattern(color->_pattern) {
```

Comment:
You might like to make this more like a copy constructor and take `const __CGColor&`